packages/bolt-connection/src/channel/browser/browser-channel.js

```diff
@@ -163,17 +164,23 @@ export default class WebSocketChannel {
    * @returns {Promise} A promise that will be resolved after channel is closed
    */
   close () {
-    return new Promise((resolve, reject) => {
-      this._clearConnectionTimeout()
-      if (this._ws && this._ws.readyState !== WS_CLOSED) {
-        this._open = false
-        this.stopReceiveTimeout()
-        this._ws.onclose = () => resolve()
-        this._ws.close()
-      } else {
-        resolve()
-      }
-    })
+    if (this._closingPromise === null) {
```

Comment:
At first glance, this change seems unrelated. Is it? 

I'm pretty sure the change is, if anything, beneficial. Yet, I'd like to understand if it was necessary or precaution and whether/how it relates to the timeout fix.

Reply:
During the tests, I noticed problems with the connection get much time to closed in browser and then other parts of the code calling the close method also, this was causing very confusing error. This code protects for this cases. 

I can move it for other PR, if it fits better.

Reply:
Sure would be cleaner to move it into a separate PR, but I don't mind sneaking in a small fix ;)

Reply:
Moved to other PR https://github.com/neo4j/neo4j-javascript-driver/pull/1169